main/modes/jumper/mode_jumper.c

```diff
@@ -0,0 +1,1125 @@
+//==============================================================================
+// Includes
+//==============================================================================
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "swadgeMode.h"
+#include "musical_buzzer.h"
+#include "esp_log.h"
+#include "esp_timer.h"
+#include "esp_random.h"
+
+#include "linked_list.h"
+#include "led_util.h"
+#include "aabb_utils.h"
+#include "settingsManager.h"
+
+#include "mode_jumper.h"
+#include "jumper_menu.h"
+
+//==============================================================================
+// Constants
+//==============================================================================
+
+#define TO_SECONDS 1000000
+
+static const uint8_t rowOffset[] = {5, 10, 15, 10, 5};
+static const float aiResponseTime[] = {2, 1, .9, .8, .7, .6, .5, .4, .3, .2}; // level 1 will have a time of 3
```

Comment:
@MrTroyD if you can, try not to use `float`. It's just so much more intensive than fixed point on ESPs. I don't think you're doing enough `float` math to have a noticeable impact, so it's mostly a me style/paranoia thing.